concept/type_/type_manager.rs

```diff
@@ -2659,144 +2348,130 @@ impl TypeManager {
-            OperationTimeValidation::validate_no_instances_to_override_relates(
-                snapshot,
-                self,
-                thing_manager,
-                relates.relation(),
-                overridden.role(),
-            )
             .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
-        }
 
-        OperationTimeValidation::validate_updated_annotations_compatible_with_role_type_and_subtypes_instances_on_supertype_change(
+        OperationTimeValidation::validate_affected_constraints_compatible_with_relates_instances_on_role_supertype_change(
             snapshot,
             self,
             thing_manager,
-            relates.role(),
-            overridden.role(),
+            subtype.clone(),
+            supertype.clone(),
         )
             .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
 
-        OperationTimeValidation::validate_updated_annotations_compatible_with_relates_and_overriding_relates_instances_on_override(
+        OperationTimeValidation::validate_affected_constraints_compatible_with_plays_instances_on_role_supertype_change(
             snapshot,
             self,
             thing_manager,
-            relates.clone(),
-            Some(overridden.clone()),
+            subtype.clone(),
+            supertype.clone(),
         )
             .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
 
-        self.set_supertype(snapshot, relates.role(), overridden.role())?;
-        TypeWriter::storage_set_type_edge_overridden(snapshot, relates, overridden);
-        Ok(())
+        self.set_supertype(snapshot, subtype, supertype)
     }
 
-    pub(crate) fn unset_relates_override(
+    pub(crate) fn unset_relates_specialise(
         &self,
         snapshot: &mut impl WritableSnapshot,
         thing_manager: &ThingManager,
         relates: Relates<'static>,
     ) -> Result<(), ConceptWriteError> {
         let role_type = relates.role();
-        if let Some(superrole_type) = role_type.get_supertype(snapshot, self)? {
-            OperationTimeValidation::validate_cardinality_of_inheritance_line_with_updated_override(
-                snapshot,
-                self,
-                relates.clone(),
-                None, // unset override
-            )
-            .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
+        if let Some(supertype) = role_type.get_supertype(snapshot, self)? {
+            self.unset_role_type_supertype(snapshot, thing_manager, role_type)?;
 
-            OperationTimeValidation::validate_interface_change_supertype_does_not_corrupt_capabilities_overrides_transitive::<Plays<'static>>(
-                snapshot,
-                relates.role(),
-                None, // new_interface_supertype
-            )
-                .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
+            if supertype.get_subtypes(snapshot, self)?.is_empty() {
```

Comment:
Here, if we unset `as role`, and it is not specialised by anyone, we automatically unset `@abstract` from `relates`. However, it could be set manually, so I'd prefer to leave it `@abstract` and ask the user to undefine `@abstract` manually, even if they haven't defined it, using only `as` instead. I've left this code here for reference. @flyingsilverfin thoughts?

A potential "clean" solution here would be to generate an `@abstract` constraint based on the fact that `relates` is specialised, and save it as we previously did. But it is indeed redundant (and I don't want to use `relates.role().get_supertype().is_some()`...)